hadoop-ozone/tools/src/main/java/org/apache/hadoop/ozone/repair/om/SnapshotRepair.java

```diff
@@ -0,0 +1,196 @@
+      ColumnFamilyHandle snapshotInfoCfh = getSnapshotInfoCfh(cfHandleList);
+      if (snapshotInfoCfh == null) {
+        System.err.println(SNAPSHOT_INFO_TABLE + " is not in a column family in DB for the given path.");
+        return null;
+      }
+
+      String snapshotInfoTableKey = SnapshotInfo.getTableKey(bucketUri.getValue().getVolumeName(),
+          bucketUri.getValue().getBucketName(), snapshotName);
+
+      SnapshotInfo snapshotInfo = getSnapshotInfo(db, snapshotInfoCfh, snapshotInfoTableKey);
+      if (snapshotInfo == null) {
+        System.err.println(snapshotName + " doesn't not exist for given bucketUri: " + OM_KEY_PREFIX +
+            bucketUri.getValue().getVolumeName() + OM_KEY_PREFIX + bucketUri.getValue().getBucketName());
+        return null;
+      }
+
+      // snapshotIdSet is the set of the all existed snapshots ID to make that the provided global previous and path
+      // previous exist and after the update snapshot doesn't point to ghost snapshot.
+      Set<UUID> snapshotIdSet = getSnapshotIdSet(db, snapshotInfoCfh);
+
+      if (Objects.equals(snapshotInfo.getSnapshotId(), globalPreviousSnapshotId)) {
+        System.err.println("globalPreviousSnapshotId: '" + globalPreviousSnapshotId +
+            "' is equal to given snapshot's ID: '" + snapshotInfo.getSnapshotId() + "'.");
+        return null;
+      }
+
+      if (Objects.equals(snapshotInfo.getSnapshotId(), pathPreviousSnapshotId)) {
+        System.err.println("pathPreviousSnapshotId: '" + pathPreviousSnapshotId +
+            "' is equal to given snapshot's ID: '" + snapshotInfo.getSnapshotId() + "'.");
+        return null;
+      }
+
+      if (!snapshotIdSet.contains(globalPreviousSnapshotId)) {
+        System.err.println("globalPreviousSnapshotId: '" + globalPreviousSnapshotId +
+            "' doesn't not exist in snapshotInfoTable.");
+        return null;
+      }
+
+      if (!snapshotIdSet.contains(pathPreviousSnapshotId)) {
+        System.err.println("pathPreviousSnapshotId: '" + pathPreviousSnapshotId +
+            "' doesn't not exist in snapshotInfoTable.");
+        return null;
+      }
+
+      if (dryRun) {
+        System.out.println("SnapshotInfo will be updated to : " + snapshotInfo);
+      } else {
+        byte[] snapshotInfoBytes = SnapshotInfo.getCodec().toPersistedFormat(snapshotInfo);
+        db.get()
+            .put(snapshotInfoCfh, StringCodec.get().toPersistedFormat(snapshotInfoTableKey), snapshotInfoBytes);
+
+        System.out.println("Snapshot Info is updated to : " +
+            getSnapshotInfo(db, snapshotInfoCfh, snapshotInfoTableKey));
+      }
+    } catch (RocksDBException exception) {
+      System.err.println("Failed to update the RocksDB for the given path: " + parent.getDbPath() +
+          "\nMake sure that Ozone entity (OM, SCM or DN) is not running for the give dbPath and current host.\n" +
+          exception);
```

Comment:
nit: wouldn't this be nicer with 3 separate `println()` calls?

Reply:
Created three separate `println()`.